### PR TITLE
Move user profile fetch and state management to redux for global access

### PR DESCRIPTION
**Pull request by mjoyce91**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,10 @@
 import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { Route, useHistory } from 'react-router-dom';
 import { Container } from 'react-bootstrap';
-import axios from 'axios';
-import { Provider } from 'react-redux';
 import { get } from 'lodash';
+import { Dispatch } from 'redux';
```
Comment (ConnorDY) [inline src/App.tsx:6]: Is this import being used?

File: src/actions/user/user.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import axios from 'axios';
+import { Dispatch } from 'redux';
+import { get } from 'lodash';
+import { SET_USER, SystemState } from '../../reducers/user/types';
+import { history } from '../../index';
+import { buildEndpointUrl } from '../../utilities';
+
+// TypeScript infers that this function is returning SendMessageAction
+const setUser = ({ user, isLoading, hasErrored } : { user?: SystemState, isLoading?: Boolean, hasErrored?: Boolean }) => {
```
Comment (ConnorDY) [inline src/actions/user/user.ts:9]: Wouldn't you typically create a different action for each of these (3 actions total)?

Comment (mjoyce91) [reply src/actions/user/user.ts:9]: I've done both approaches, but I find this way to be cleaner

File: src/App.tsx (hunk 1 above)
Comment (mjoyce91) [reply src/App.tsx:6]: Nope, think I was using as a type initially. I'll remove

File: src/actions/user/user.ts (hunk 2 above)
Comment (ConnorDY) [reply src/actions/user/user.ts:9]: Doesn't this make it more difficult to use the `useEffect` hook on the state?
You can't check if only a part of the state object has been updated.

Comment (mjoyce91) [reply src/actions/user/user.ts:9]: I haven't use this approach with hooks, but you should still be able to do something like:

```jsx
const nestedProp = useSelector((state: RootState) => state.user.someNestedProp)
```

Comment (ConnorDY) [reply src/actions/user/user.ts:9]: I'm curious if
```javascript
useEffect(() => {
  // do something
}, [nestedProp]);
```
will work as expected. I think that will determine if we can use this approach or not.

**Comment by mjoyce91** [comment]
@ConnorDY We can hold off on this until necessary

**Comment by ConnorDY** [comment]
> @ConnorDY We can hold off on this until necessary

I'm not sure if this app is complex enough to need Redux.

**Comment by mjoyce91** [comment]
Closing for now

**Pull request closed without merging**